src/main/java/net/minecraftforge/event/TickEvent.java

```diff
@@ -38,78 +38,147 @@ public static class ServerTickEvent extends TickEvent {
         private final BooleanSupplier haveTime;
         private final MinecraftServer server;
 
-        public ServerTickEvent(Phase phase, BooleanSupplier haveTime, MinecraftServer server)
-        {
+        protected ServerTickEvent(BooleanSupplier haveTime, MinecraftServer server, Phase phase) {
```

Comment:
All of these public -> protected are binary breaks.
Its fine to change them to protected {and move the phase to the end} but the public ones have to stay as bouncers. 
And also you should mark the proper time for removal
`    @Deprecated(forRemoval = true, since = "1.20.6")`

Maintaining binary compatibility means that your PR can be backported without major modifications
```suggestion
        @Deprecated(forRemoval = true, since = "1.20.6")
        public ServerTickEvent(Phase phase, BooleanSupplier haveTime, MinecraftServer server) {
            this(haveTime, server, phase);
        }
        protected ServerTickEvent(BooleanSupplier haveTime, MinecraftServer server, Phase phase) {
```

Reply:
Ah yes I completely missed that, thank you! I think I managed to get all of them now